infra/aws/terraform/prow-build-cluster/resources/external-secrets/external-secrets.yaml

```diff
@@ -17,6 +17,8 @@ apiVersion: v1
 kind: ServiceAccount
 metadata:
   name: external-secrets
+  annotations:
+    eks.amazonaws.com/role-arn: arn:aws:iam::054318140392:role/SECRETSMANAGER-IRSA
```

Comment:
This is going to be a problem because account ID is different for prod and canary.

Reply:
Yep, that's why we need to modify flux setup a bit. I was thinking about creating something similar to https://github.com/fluxcd/flux2-kustomize-helm-example